File: src/main/java/de/aquafun3d/bingo/Listeners/NewItemListener.java
package de.aquafun3d.bingo.Listeners;

import net.kyori.adventure.text.Component;
import org.bukkit.ChatColor;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityPickupItemEvent;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.inventory.ItemStack;

public class NewItemListener implements Listener {


    public NewItemListener(){

    }

    @EventHandler
    public void onInvClick(InventoryClickEvent e){
        Player player = (Player) e.getWhoClicked();
        ItemStack item = e.getCurrentItem();
        if (e.getView().title().equals(Component.text(ChatColor.DARK_PURPLE + "Teamselect"))){
            return;
        }
        if (e.getView().title().equals(Component.text(ChatColor.DARK_PURPLE + "Bingo"))){
            e.setCancelled(true);
            return;
        }
        if (item == null) {
            return;
        }
        if (item.getItemMeta() == null) {
            return;
        }
        checkItem(player, item);
    }

    @EventHandler
    public void onPickUp(EntityPickupItemEvent e) {
        if (e.getEntity() instanceof Player player) {
            ItemStack item = e.getItem().getItemStack();
            checkItem(player, item);
        }
    }

    public void checkItem(Player player, ItemStack item){
        //TODO
    }

    public void remove(Player player, ItemStack item){
        //TODO
    }
}

File: src/main/java/de/aquafun3d/bingo/listeners/NewItemListener.kt
package de.aquafun3d.bingo.listeners

import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.NamedTextColor
import org.bukkit.*
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityPickupItemEvent
import org.bukkit.event.inventory.InventoryClickEvent
import org.bukkit.inventory.ItemStack

class NewItemListener : Listener {
    @EventHandler
    fun onInvClick(e: InventoryClickEvent) {
        val player = e.whoClicked as Player
        val item = e.currentItem
        if (e.view.title() == Component.text("Teamselect", NamedTextColor.DARK_PURPLE)) {
            return
        }
        if (e.view.title() == Component.text("Bingo", NamedTextColor.DARK_PURPLE)) {
            e.isCancelled = true
            return
        }
        if (item == null) {
            return
        }
        if (item.itemMeta == null) {
            return
        }
        //checkItem(player, item)
    }

    @EventHandler
    fun onPickUp(e: EntityPickupItemEvent) {
        if (e.entity is Player){
            val player = (e.entity as Player)
            val item = e.item.itemStack
            //checkItem(player, item)
        }
    }
}
